src/mds/MDCache.cc

```diff
@@ -1418,6 +1428,34 @@ void MDCache::adjust_subtree_after_rename(CInode *diri, CDir *olddir, bool pop)
   show_subtrees();
 }
 
+// ===================================
+// Cache Autotuning
+
+
+/*
+ * If mds_cache_autotune is set to true, then adjust cache_memory_limit using the rss usage periodically as per the mds_cache_resize_interval
+ */
+void MDCache::resize()
+{
+    if (cache_autotune) {
+      auto now = clock::now();
+      auto cache_resize_interval = g_conf().get_val<std::chrono::seconds>("mds_cache_resize_interval"); 
+      if (cache_memory_limit > mds->get_mds_memory_target())
+        // RSS usage has been experimentally shown to be around 25 percent more than the cache_memory_limit. Hence when the mds_memory_target is changed and the cache_memory_limit becomes more than the memory_target, the cache_memory_limit is reset as per the mds_memory_target(which is an upper bound on the rss usage) 
+        cache_memory_limit = (mds->get_mds_memory_target()*100)/125;
+      if (cache_resize_interval == std::chrono::duration<long int>(0) || ((now-last_cache_resize_time) > cache_resize_interval)) {
+        MemoryModel::snap last;
+        mm.sample(&last);
+        if (((uint64_t)(last.get_rss())) > mds->get_mds_memory_target()) 
+          cache_memory_limit = (cache_memory_limit/100)*99;
+        else if (!cache_overfull() && ((uint64_t)(last.get_rss())) < (mds->get_mds_memory_target()/100) * 98)
+          cache_memory_limit = (cache_memory_limit/100) * 101;
```

Comment:
Whenever we change the `cache_memory_limit`, we should note this in the MDS log. I suggest `dout(5)`.